### PR TITLE
Shearwater cloud

### PR DESCRIPTION
**Pull request by mturkia**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Fix a few bugs in Shearwater Cloud import. Take also into account some recent changes in the Shearwater database format.

### Changes made:

1) Fix cylinder handling
2) Fix gas amount import
3) Take into account bogus data in Shearwater database (null rows)
4) Switch to use timestamps in milliseconds that has been introduced by Shearwater

### Release note:

Line added

### CODE REVIEW

File: xslt/shearwater.xslt [hunk 1]
```diff
@@ -60,6 +60,17 @@
         <xsl:value-of select="concat(maxTime, ' min')"/>
       </xsl:attribute>
 
+      <xsl:variable name="timeMultiplier">
+        <xsl:choose>
+          <xsl:when test="diveLogRecords/diveLogRecord[1]/currentTime &gt; 100">
```
Comment (dirkhh) [inline xslt/shearwater.xslt:65]: I think that makes sense. if your sample rate is faster than 100ms or slower than 100s, you have other issues as well.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,18 @@
+import: fix a copule of bugs on Shearwater Cloud import
```
Comment (dirkhh) [inline CHANGELOG.md:1]: you are bringing the old CHANGELOG back :-)
No worries, I'll fix this in post 🤣

**Review by dirkhh — APPROVED** [review]
This looks good. I mean, it looks horrible, all these weird hacks to fix things that are broken or odd in the output of someone else's tools which is inherently unstable and also XSLT and GAH!!!
But yeah, the PR makes sense and shouldn't affect anyone else.
I'll deal with the CHANGELOG issue by merging locally and just adding this to the ReleaseNotes instead

**Comment by dirkhh** [comment]
manually merged

**Comment by mturkia** [comment]
Where was this merged? I don't see it in subsurface/master

**Comment by dirkhh** [comment]
Ugh. Did I not push the local merge? That sounds like something I would do these days.
Yep, that's what I did.
Just pushed it out.
Thank you so much for catching it!

**Pull request closed without merging**